src/Files/NoStdio.class.st

```diff
@@ -0,0 +1,41 @@
+"
+I am NoStdio. I am a Stdio that does not create files for windows but just streams stored in memory. 
+This is an hack due to the missing handles in the VM for Windows.
+
+I am used to set up an image to run an image without writing files on disk.
+
+To install me, execute
+
+  NoStdio install.
+
+To get back to the normal situation, execute
+
+  Stdio reset.
+"
+Class {
+	#name : #NoStdio,
+	#superclass : #Stdio,
+	#category : #'Files-Core'
+}
+
+{ #category : #singleton }
+NoStdio class >> install [
+	DefaultInstance := self new
+]
+
+{ #category : #stdio }
+NoStdio >> standardIOStreamNamed: moniker forWrite: forWrite [
+	"Create if necessary and store default stdin, stdout and other files based on the their names"
+
+	| handle |
+	self flag: #todo.	"This is an ugly hack, while waiting for a real fix for windows. There are several problems with this approach, but it allows us to run tests, etc."
+	Smalltalk os isWin32
```

Comment:
As far as I know we have two versions of Windows VM. The one with the suffix "Console"  should be able to use standard streams. So then this platform check will fail.

Reply:
Right... But how can I know that is a console or not?

Reply:
you can't, that's an important part of the problem (and why this is not solved long before). Only real way to know is to open the stdout stream and try to use it. If it fails, is not a console :(

Reply:
back to this. 
how do we solve this? (checking by binary name seems a weak strategy). I guess best is to add a primitive that can answer this (shame, but I guess needed).

what do you think?

Reply:
Hi Vincent,
I'm missing what the goal of this is.

At the moment it appears that both FileStream and Stdio force stdio streams to be written to regular files on Windows - I'm not sure why.

If that code is removed, it is possible to write to the console with PharoConsole.exe using FileStream>>stdout.  When I tried it with Stdio>>stdout it didn't produce anything.  I haven't investigated further.

But...

With PharoConsole.exe:

FileStream stdout closed 
" false"


With Pharo.exe:

FileStream stdout closed
" true"

Does that help?

Cheers,
Alistair